src/test/java/org/jenkinsci/plugins/workflow/cps/DSLTest.java

```diff
@@ -420,6 +433,176 @@ public void namedSoleParamForStep() throws Exception {
+        WorkflowRun run = r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+        r.assertLogContains("Warning: A secret was passed to \""+ shellStep + "\"", run);
+        r.assertLogContains("Affected argument(s) used the following variable(s): [PASSWORD]", run);
+        InterpolatedSecretsAction reportAction = run.getAction(InterpolatedSecretsAction.class);
+        Assert.assertNotNull(reportAction);
+        List<InterpolatedSecretsAction.InterpolatedWarnings> warnings = reportAction.getWarnings();
+        MatcherAssert.assertThat(warnings.size(), is(1));
+        InterpolatedSecretsAction.InterpolatedWarnings stepWarning = warnings.get(0);
+        MatcherAssert.assertThat(stepWarning.getStepSignature(), is(shellStep + "(script: echo ${PASSWORD})"));
+        MatcherAssert.assertThat(stepWarning.getInterpolatedVariables(), is(Arrays.asList("PASSWORD")));
+        LinearScanner scan = new LinearScanner();
+        FlowNode node = scan.findFirstMatch(run.getExecution().getCurrentHeads().get(0), new NodeStepTypePredicate(shellStep));
+        ArgumentsAction argAction = node.getPersistentAction(ArgumentsAction.class);
+        Assert.assertFalse(argAction.isUnmodifiedArguments());
+        MatcherAssert.assertThat(argAction.getArguments().values().iterator().next(), is("echo ${PASSWORD}"));
+    }
+
+    @Issue("JENKINS-63254")
+    @Test public void sensitiveVariableInterpolationWithMetaStep() throws Exception {
+        final String credentialsId = "creds";
+        final String username = "bob";
+        final String password = "secr3t";
+        UsernamePasswordCredentialsImpl c = new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL, credentialsId, "sample", username, password);
+        CredentialsProvider.lookupStores(r.jenkins).iterator().next().addCredentials(Domain.global(), c);
+        p.setDefinition(new CpsFlowDefinition(""
+                + "node {\n"
+                + "withCredentials([usernamePassword(credentialsId: 'creds', usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]) {\n"
+                + "archiveArtifacts(\"${PASSWORD}\")"
+                + "}\n"
+                + "}", true));
+        WorkflowRun run = r.assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0));
+        r.assertLogContains("Warning: A secret was passed to \"archiveArtifacts\"", run);
+        r.assertLogContains("Affected argument(s) used the following variable(s): [PASSWORD]", run);
+        InterpolatedSecretsAction reportAction = run.getAction(InterpolatedSecretsAction.class);
+        Assert.assertNotNull(reportAction);
+        List<InterpolatedSecretsAction.InterpolatedWarnings> warnings = reportAction.getWarnings();
+        MatcherAssert.assertThat(warnings.size(), is(1));
+        InterpolatedSecretsAction.InterpolatedWarnings stepWarning = warnings.get(0);
+        MatcherAssert.assertThat(stepWarning.getStepSignature(), is("archiveArtifacts(delegate: archiveArtifacts(<anonymous>: ${PASSWORD}))"));
```

Comment:
This test and a few others shows there are still a few issues with `InterpolatedWarnings.getStepSignature`. It would be good to fix these, but I also think they are relatively unlikely to occur in practice with interpolated passwords, so may it would be better to merge this and handle that in a follow-up?

1. When using symbol-based syntax and the `UninstantiateDescribable` only has one required parameter and there is only one provided argument (i.e. [`hasSoleRequiredParameter`](https://github.com/jenkinsci/structs-plugin/blob/745286d2d7a362a6b39579073fb2f01bb0f5c3b8/plugin/src/main/java/org/jenkinsci/plugins/structs/describable/UninstantiatedDescribable.java#L107) returns true), the value can be provided directly instead of using the `key: value` syntax, meaning that we should never see things like `<anonymous>:` in the output. In this particular case, fixing this issue (but not the other issue) would make the output `archiveArtifacts(archiveArtifacts(${PASSWORD}))`, which is a bit closer to what was written in the Pipeline.
2. Meta steps like `archiveArtifacts` are handled incorrectly. For example, in this case the outer `archiveArtifacts` is actually the [`step`](https://github.com/jenkinsci/workflow-basic-steps-plugin/blob/e3e6f75aece39b1f6698637ab7a5c0c05f2cfc21/src/main/java/org/jenkinsci/plugins/workflow/steps/CoreStep.java#L59) step, but generally the outer step is should be omitted so only the inner describable is shown. The direct meta step invocation in `DSL.invokeDescribable` specifically exists to handle this case so users don't have to write use `step` explicitly. You can  check for this case using `StepDescriptor.isMetaStep` (check that that `FlowNode` is an instance of `StepNode` and then call `StepNode.getDescriptor`) along with making sure that the argument is for the `Describable` type corresponding to the required parameter of the meta step, maybe with something like this in `InterpolatedWarnings.getStepSignature`:

    ```
    FlowNode node; // Maybe return this when looking up step arguments?
    if (node instanceof StepNode) {
        StepDescriptor descriptor = ((StepNode)node).getDescriptor();
        if (descriptor.isMetaStep()) {
            DescribableParameter p = DescribableModel.of(descriptor.clazz).getFirstRequiredParameter();
            if (p != null) {
                Object arg = ArgumentsAction.getResolvedArguments(f).get(p.getName());
                return argumentToString(arg);
            }
        }
    }
    // handle as normal step using existing code
    ```

Reply:
Attempted to address comments in 7d8672c